### DISCUSSION TITLE
Your configFile is invalid: tests/cypress/cypress.config.js - TS install

### DISCUSSION THREAD

**Discussion by Mehdi-Skills** [cypress-io/cypress#27441] [Questions and Help]
Hello, 

I'm looking to implement cypress on an existing front-end project. 

I'm encountering an error with the config File, which is invalid. 

I've installed Cypress as recommended in the documentation. 

Here's my file tree: 
![Capture d’écran 2023-08-02 à 10 28 53](https://github.com/cypress-io/cypress/assets/125974014/f0b1898f-ba16-4c4c-8e9d-6162c0ab74b3)

I have the following typescript configuration file tsconfig.json at the root of the project: 
![Capture d’écran 2023-08-02 à 10 34 30](https://github.com/cypress-io/cypress/assets/125974014/31893e49-0042-44b3-b5bf-193e8eef7d22)

Here's my cypress.config.js file in the cypress folder: 
![Capture d’écran 2023-08-02 à 10 55 09](https://github.com/cypress-io/cypress/assets/125974014/23cdb150-ebe3-45ea-8013-1705da8a35d0)

The cypress error is strange here's the error reported on Cypress when opening via a "npx cypress open": 
![Capture d’écran 2023-08-02 à 10 35 42](https://github.com/cypress-io/cypress/assets/125974014/f3ecd36d-db71-459e-9ef4-076b29a225b2)

If I put the tsconfig.json include like this : 
<img width="525" alt="Capture d’écran 2023-08-02 à 11 02 30" src="https://github.com/cypress-io/cypress/assets/125974014/d97f3a8a-99ee-4340-b057-967609ae729b">

I have an error in my tsconfig file : Unknown compiler option 'include'.ts 

but when I relaunch cypress via a "npx cypress open",  I get this message: 
<img width="934" alt="Capture d’écran 2023-08-02 à 11 04 07" src="https://github.com/cypress-io/cypress/assets/125974014/ddb9d708-a3d3-4ecd-b2d1-a7045cb406c0">

I don't understand and I think my problem is a file configuration problem with TS, but I'm stuck. 

Thank you in advance for your help

**Comment by xinxiong-tw** [comment]
Hey, I just encountered the same problem, and I found out that it's because the 'module' configuration in tsconfig.json file is set to 'commonjs', so I change it to 'ESNext', and it's done.
Mabey you can change the module configuration and try again

**Reply by atorscho** [reply]
Thank you! That fixed it
